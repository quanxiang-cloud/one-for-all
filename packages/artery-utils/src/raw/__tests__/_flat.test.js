import { fromJS } from 'immutable';
import flat from '../_flat';

import NODE from './fixtures/node';

test('flat', () => {
  const pairs = flat(fromJS(NODE));

  const idPairs = pairs.map(([keyPath, node]) => {
    return [keyPath.toJS().join('/'), node.getIn(['id'])];
  });

  expect(idPairs).toMatchSnapshot();
});
