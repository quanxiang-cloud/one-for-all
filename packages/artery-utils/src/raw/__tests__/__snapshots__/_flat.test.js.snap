// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flat 1`] = `
Array [
  Array [
    "",
    "container",
  ],
  Array [
    "children/0",
    "ref-schema",
  ],
  Array [
    "children/1",
    "todo-input-container",
  ],
  Array [
    "children/1/children/0",
    "todo-input",
  ],
  Array [
    "children/2",
    "todo-list-loop-composedNode",
  ],
  Array [
    "children/2/node",
    "compose-node-container",
  ],
  Array [
    "children/2/node/nodes/0",
    "todo-toggle",
  ],
  Array [
    "children/2/node/nodes/1",
    "todo-title",
  ],
  Array [
    "children/2/node/nodes/2",
    "todo-delete",
  ],
  Array [
    "children/3",
    "footer",
  ],
  Array [
    "children/3/children/0",
    "todo-filter",
  ],
]
`;
